lib/amqp.js

```diff
@@ -284,10 +284,14 @@ class Amqp {
         }
         const errorPayload = JSON.stringify(payload);
 
-        let result = this.sendToExchange(settings.PUBLISH_MESSAGES_TO, settings.ERROR_ROUTING_KEY,
-            errorPayload, properties, throttle);
+        let result = this.sendToExchange(
+            settings.PUBLISH_MESSAGES_TO,
+            settings.ERROR_ROUTING_KEY,
+            errorPayload, properties,
+            throttle
+        );
 
-        if (headers.reply_to) {
+        if (!settings.NO_ERROR_REPLIES && headers.reply_to) {
```

Comment:
Think about: probably this logic should live in sailor.js
Sailor.js => semantics
Amqp.js => transport layer.